### PR TITLE
Upgrade GitHub Insights plugin version in sample app

### PR DESCRIPTION
**Pull request by lewtakm**
Upgrade from 0.2.2 to 0.2.6 that includes some fixes regarding private GitHub repositories. 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by dtuite** [comment]
This update addresses an issue reported by @andrewthauer last night [on Discord](https://discord.com/channels/687207715902193673/687207715902193679/765929726576427049).

When the GitHub insights plugin was installed, private repos would throw an error like the below error and refuse to render in the catalog. This is fixed in `@roadiehq/backstage-plugin-github-insights` version `0.2.5` and greater.

```
TypeError: can't access property "color", colors[language[0]] is undefined
    LanguagesCard index.esm.js:1925
    LanguagesCard index.esm.js:1913
    renderWithHooks react-dom.development.js:14826
    updateFunctionComponent react-dom.development.js:17059
    beginWork react-dom.development.js:18644
    callCallback react-dom.development.js:189
    invokeGuardedCallbackDev react-dom.development.js:238
    invokeGuardedCallback react-dom.development.js:293
    beginWork$1 react-dom.development.js:23235
    performUnitOfWork react-dom.development.js:22186
    workLoopSync react-dom.development.js:22162
    performSyncWorkOnRoot react-dom.development.js:21788
    flushSyncCallbackQueueImpl react-dom.development.js:11112
    unstable_runWithPriority scheduler.development.js:653
    runWithPriority$1 react-dom.development.js:11062
    flushSyncCallbackQueueImpl react-dom.development.js:11107
    flushSyncCallbackQueue react-dom.development.js:11095
    scheduleUpdateOnFiber react-dom.development.js:21231
    dispatchAction react-dom.development.js:15683
    callback useAsyncFn.js:18
react-hot-loader.development.js:294
```

To prevent issues like this in the future, we will do the following:

 1. Increase our use of React error boundaries in order to prevent these issues spilling out of the plugin.
 2. Do more validation of request responses so that mismatched response shapes don't cause such uncontrolled issues.

File: yarn.lock [hunk 1]
```diff
@@ -3690,29 +3708,6 @@
     react-router "6.0.0-beta.0"
     react-use "^15.3.3"
 
-"@roadiehq/backstage-plugin-travis-ci@^0.2.3":
```
Comment (freben) [inline yarn.lock:3693]: Hm. I wonder why this block got removed.

**Review by freben — APPROVED** [review]
Tentatively approved, but could you double check that you ran this on a clean newest master before changing those lines? There's a bit more changed in yarn.lock than I'd expected.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Gonna go a bit further than freben and make sure we don't merge this until the `yarn.lock` changes are sorted out. Any idea where they are coming from?

**Comment by dtuite** [comment]
@freben @Rugvip  Yeah I thought the same. We are looking.

**Comment by Rugvip** [comment]
> 1. Increase our use of React error boundaries in order to prevent these issues spilling out of the plugin.

We plan to add this as part of the core API, so feel free to skip this or slap a `TODO: remove` on it.

**Review by Rugvip — APPROVED** [review]

**Comment by dtuite** [comment]
Cleaner lockfile committed.

@Rugvip one other idea. Dependabot is ignoring our plugin updates because they are `^0.x.y`. Only major upgrades are handled automatically. If we replace the caret with a tilde, my understanding is that minor upgrades will be automatically handled by Dependabot. Thoughts? 

[NPM docs for tilde and caret handling](https://docs.npmjs.com/misc/semver#tilde-ranges-123-12-1).

**Review by freben — APPROVED** [review]
That's a lot better! I'm leaving for the day, somebody else may be able to help with the merge once it builds

**Comment by Rugvip** [comment]
@dtuite I think dependabot bumps regardless of whether it's within the specified range or not, e.g. https://github.com/spotify/backstage/pull/2668/files and https://github.com/spotify/backstage/pull/2912/files. I think the reason it's now showing up is because dependabot posts a limited number PRs and we need to take care of the existing ones first.

**Comment by dtuite** [comment]
@Rugvip Ah. That makes sense. Thanks

**Comment by dtuite** [comment]
@andrewthauer Since you reported this, want to have the privilege of hitting the big green button? 😄

**Pull request merged**